activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb

```diff
@@ -115,7 +115,11 @@ def initialize(...)
         end
 
         @config[:strict] = ConnectionAdapters::SQLite3Adapter.strict_strings_by_default unless @config.key?(:strict)
-        @connection_parameters = @config.merge(database: @config[:database].to_s, results_as_hash: true)
+        @connection_parameters = @config.merge(
+          database: @config[:database].to_s,
+          results_as_hash: true,
+          default_transaction_mode: :deferred
```

Comment:
@fractaledmind Have you perhaps mistakenly put it back to `:deferred` instead of `:immediate` when splitting into multi lines?

Reply:
Whoops. A reason for not committing after 10pm... Thanks for the pointer. Fixing.